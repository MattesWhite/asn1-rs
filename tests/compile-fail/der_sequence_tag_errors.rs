fn test_tag_errors() {
    use asn1_rs::*;

    /// Should not compile: both implicit and explicit tags
    #[derive(Debug, PartialEq, DerSequence)]
    // #[debug_derive]
    pub struct T0 {
        #[tag_explicit(0)]
        #[tag_implicit(0)]
        a: u16,
    }

    /// Should not compile: multiple explicit tags
    #[derive(Debug, PartialEq, DerSequence)]
    // #[debug_derive]
    pub struct T1 {
        #[tag_explicit(0)]
        #[tag_explicit(1)]
        a: u16,
    }
}

fn main() {
    test_tag_errors();
}
