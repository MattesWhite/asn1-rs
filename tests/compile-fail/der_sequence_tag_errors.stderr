error: proc-macro derive panicked
 --> tests/compile-fail/der_sequence_tag_errors.rs:5:32
  |
5 |     #[derive(Debug, PartialEq, DerSequence)]
  |                                ^^^^^^^^^^^
  |
  = help: message: tag cannot be set twice!

error: proc-macro derive panicked
  --> tests/compile-fail/der_sequence_tag_errors.rs:14:32
   |
14 |     #[derive(Debug, PartialEq, DerSequence)]
   |                                ^^^^^^^^^^^
   |
   = help: message: tag cannot be set twice!
